src/send-reminders.js

```diff
@@ -0,0 +1,161 @@
+import pMap from 'p-map'
+import VError from 'verror'
+import { serializeError } from 'serialize-error'
+
+let apiRoot
+let ctpClient
+let logger
+let stats
+let activeStateId
+
+async function sendReminders({
+  apiRoot: _apiRoot,
+  ctpClient: _ctpClient,
+  logger: _logger,
+}) {
+  stats = {
+    processedTemplateOrders: 0,
+    updatedTemplateOrders: 0,
+    skippedTemplateOrders: 0,
+  }
+
+  try {
+    apiRoot = _apiRoot
+    ctpClient = _ctpClient
+    logger = _logger
+
+    await _setActiveStateId()
+
+    const orderQuery =
+      await _buildQueryOfTemplateOrdersThatIsReadyToSendReminder(activeStateId)
+
+    for await (const templateOrders of ctpClient.fetchPagesGraphQl(orderQuery))
+      await pMap(templateOrders, _processTemplateOrder, { concurrency: 3 })
+
+    return stats
+  } catch (err) {
+    logger.error(
+      'Failed on send reminders, processing should be restarted on the next run.' +
+        `Error: ${JSON.stringify(serializeError(err))}`
+    )
+    return stats
+  }
+}
+
+async function _setActiveStateId() {
+  const {
+    body: { id },
+  } = await apiRoot.states().withKey({ key: 'Active' }).get().execute()
+  activeStateId = id
```

Comment:
Maybe return it here instead and do not use class variable? I see you're passing it anyway as a method argument. And rename the method to `_fetchActiveStateId`

Reply:
https://github.com/commercetools/commercetools-subscriptions/pull/8/commits/df80a90c771ee02332cc2cf95d5d5877c0e3e2bd